### DISCUSSION TITLE
OWS and App Settings

### DISCUSSION THREAD

**Discussion by tstallan** [vanjarosoftware/Vanjaro.Platform#2174] [Q&A]
Hi,

I use the OWS module extensively for to display/edit anything to do data. It works great in all versions of DNN and the module can be placed onto a page using Vanjaro.

However, once the module is on page I cannot get to the settings to tell it what OWS code package to use. In normal DNN opening the module settings shows an additional 'OWS Settings' tab where I can select the code package.

Using Vanjaro I don't get this option and selecting App Settings gives me errors in the console mainly relating to dnnModal not being defined.

Any suggestions as I would love to use Vanjaro 

Thanks

Tony

**Comment by DorianKunzler-SC** [comment]
I don't have an answer for this one, but I am experiencing the same issue and error as Tony when trying to access the module settings for a HotCakes Ecommerce module.  The console show a similar error with the dnnModal.show not being found.  This is using the most recent 1.4 release of Vanjaro and DNN 9.10.2.

- Dorian

**Comment by meetmandeep** [comment]
Are you using Vanjaro Platform or Vanjaro For DNN Extension?

**Reply by DorianKunzler-SC** [reply]
I am using the Vanjaro for DNN Extension.

**Comment by tstallan** [comment]
Hi Mandeep,

I too am using the extension

Regards

Tony

**Comment by nolansa** [comment]
A quick fix for this is to replicate the functionality for removing the dnnModal Javascript code found inside the Vanjaro Container Base.ascx.cs file inside the front-end "settings.js" javascript file.

To do this open up DesktopModules/Vanjaro/UXManager/Extensions/Apps/ModuleSettings/Views/Settings/Settings.js

and add the following code to line 64:

`if ($scope.ui.data.AppSettingUrl.Value.indexOf('javascript:') > -1) {
						$scope.ui.data.AppSettingUrl.Value = $scope.ui.data.AppSettingUrl.Value.replace("javascript:dnnModal.show('", "").split(',')[0].trim('\'');
					}`

so the final code block should look like this line 63 - 69 :

`else if (type === 'apps') {
					if ($scope.ui.data.AppSettingUrl.Value.indexOf('javascript:') > -1) {
						$scope.ui.data.AppSettingUrl.Value = $scope.ui.data.AppSettingUrl.Value.replace("javascript:dnnModal.show('", "").split(',')[0].trim('\'');
					}
            window.parent.OpenPopUp(null, '100%', '', 'App Settings', $scope.ui.data.AppSettingUrl.Value, '', '', false, true);
            setTimeout(function () { window.location.reload(); }, 500);
        }`
